graphene_plugin.py

```diff
@@ -764,19 +756,36 @@ def process_gql_schema(ctx: AttributeContext) -> Type:
                         )
                         continue
 
-            # Check the every resolver function has a corresponding `Field()` defintion
-            missing_field_names = set(object_info.resolvers.keys()) - set(object_info.fields.keys())
-            for name in missing_field_names:
-                if isinstance(object_info, InterfaceInfo) and name == 'type':
-                    # This is not a field resolver. It is the special `Interface` resolver that determines which
-                    # `ObjectType` to use at runtime.
+            # If no resolver function is defined, type-check the behavior of the graphene default resolver
+            if isinstance(object_info, ObjectTypeInfo):
+                # The default resolver doesn't apply to `Interface`s because the `ObjectType`s that implement them could
+                # have resolvers for their fields.
+                # TODO: Detect if any of an `Interface`'s `ObjectType`s do _not_ define their own resolver for this
+                # field. In that case, we _do_ want to type-check the default resolver.
+                fields_without_resolver_names = set(object_info.fields.keys()) - set(object_info.resolvers.keys())
+                for name in fields_without_resolver_names:
+                    gql_field = object_info.fields[name]
+                    # Note: `analyze_member_access` will call `ctx.api.fail()` if the provided type doesn't have
+                    # a member with the given name at all. So our code only needs to do the subtype check.
+                    default_resolver_return_type = analyze_member_access(
+                        gql_field.name,
+                        object_info.runtime_type,
+                        gql_field.context,
+                        False,  # is_lvalue
+                        False,  # is_super
+                        False,  # is_operator
+                        ctx.api.msg,
+                        original_type=object_info.runtime_type,
```

Comment:
What is the difference between the `original_type` kwarg and the second arg to this fn? (I noticed you're sending the same value for both)

Reply:
From the mypy source:

> 'original_type' is the most precise inferred or declared type of the base object
    that we have available. When looking for an attribute of 'typ', we may perform
    recursive calls targeting the fallback type, and 'typ' may become some supertype
    of 'original_type'. 'original_type' is always preserved as the 'typ' type used in
    the initial, non-recursive call. The 'self_type' is a component of 'original_type'
    to which generic self should be bound (a narrower type that has a fallback to instance).
    Currently this is used only for union types.

I think I still lack the context to understand this 100% but I think the right thing to do for our usecase is just pass the same thing.